src/pydicom/pixels/utils.py

```diff
@@ -1738,6 +1738,188 @@ def reshape_pixel_array(ds: "Dataset", arr: "np.ndarray") -> "np.ndarray":
     return arr
 
 
+def set_pixel_data(
+    ds: "Dataset",
+    arr: "np.ndarray",
+    photometric_interpretation: str,
+    bits_stored: int,
+) -> None:
+    """Use an :class:`~numpy.ndarray` to set a dataset's *Pixel Data* and related
+    Image Pixel module elements.
+
+    .. versionadded:: 3.0
+
+    The following :dcm:`Image Pixel<part03/sect_C.7.6.3.3.html#table_C.7-11c>`
+    module elements values will be added, updated or removed as necessary:
+
+    * (0028,0002) *Samples per Pixel* using a value corresponding to
+      `photometric_interpretation`.
+    * (0028,0004) *Photometric Interpretation* from `photometric_interpretation`.
+    * (0028,0006) *Planar Configuration* will be added and set to ``0`` if
+      *Samples per Pixel* is > 1, otherwise it will be removed.
```

Comment:
I dimly remember somebody writing about setting an ndarray with separate color planes, though I'm not sure if this a real use case. Can be added later if needed...